src/components/new-safe/steps/Step3/index.tsx

```diff
@@ -0,0 +1,157 @@
+    return {
+      owners: data.owners.map((owner) => owner.address),
+      threshold: data.threshold,
+      saltNonce,
+    }
+  }, [data.owners, data.threshold, saltNonce])
+
+  const { gasLimit } = useEstimateSafeCreationGas(safeParams)
+
+  const totalFee =
+    gasLimit && maxFeePerGas && maxPriorityFeePerGas
+      ? formatVisualAmount(maxFeePerGas.add(maxPriorityFeePerGas).mul(gasLimit), chain?.nativeCurrency.decimals)
+      : '> 0.001'
+
+  const formMethods = useForm<CreateSafeStep3Form>({
+    mode: 'all',
+    defaultValues: {
+      [CreateSafeStep3Fields.name]: data.name,
+      [CreateSafeStep3Fields.owners]: data.owners,
+      [CreateSafeStep3Fields.threshold]: data.threshold,
+    },
+  })
+
+  const { handleSubmit, watch } = formMethods
+
+  const allFormData = watch()
+
+  const handleBack = () => {
+    onBack(allFormData)
+  }
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)} id={STEP_3_FORM_ID}>
+      <FormProvider {...formMethods}>
+        <Grid container spacing={3}>
+          <Grid item>
+            <Grid container spacing={3}>
+              <ReviewRow name="Network" value={<ChainIndicator chainId={chain?.chainId} inline />} />
+              <ReviewRow name="Name" value={<Typography>{data.name}</Typography>} />
+              <ReviewRow
+                name="Owners"
+                value={
+                  <Box className={css.ownersArray}>
+                    {data.owners.map((owner, index) => (
+                      <EthHashInfo address={owner.address} shortAddress={false} showPrefix={false} key={index} />
+                    ))}
+                  </Box>
+                }
+              />
+              <ReviewRow
+                name="Threshold"
+                value={
+                  <Typography>
+                    {data.threshold} out of {data.owners.length} owner(s)
+                  </Typography>
+                }
+              />
+            </Grid>
+          </Grid>
+
+          <Grid item xs={12}>
+            <Divider sx={{ ml: '-52px', mr: '-52px', mb: 4, mt: 3 }} />
+            <Grid item xs={12}>
+              <Grid container spacing={3}>
+                <ReviewRow
+                  name="Est. network fee"
+                  value={
+                    <Box p={1} sx={{ backgroundColor: 'secondary.background', width: 'fit-content' }}>
+                      <Typography variant="body1">
+                        <b>
+                          &asymp; {totalFee} {chain?.nativeCurrency.symbol}
+                        </b>
+                      </Typography>
+                    </Box>
+                  }
+                />
+                <ReviewRow
+                  name=""
+                  value={
+                    <Typography color="text.secondary">
+                      You will have to confirm a transaction with your currently connected wallet.
+                    </Typography>
+                  }
+                />
```

Comment:
The top spacing of this row is different than in the design. I suggest setting it to 8px.